### ISSUE TITLE
Error from an array of tags is cryptic.

### ISSUE THREAD

**Issue by mjhm** [cucumber/cucumber-js#681]
After upgrading I get the error below.  It wasn't too hard to track down, since I read the changelog, but better error message would be more helpful.

It might be worthwhile to continue to allow arrays of tags and maybe just print a deprecation warning?
```
/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/cli/run.js:21
            };cli = new _2.default({ argv: process.argv, cwd: process.cwd(), stdout: process.stdout });success = void 0;_context.prev = 3;_context.next = 6;return cli.run();case 6:success = _context.sent;_context.next = 13;break;case 9:_context.prev = 9;_context.t0 = _context['catch'](3);process.nextTick(function () {throw _context.t0;});return _context.abrupt('return');case 13:exitCode = success ? 0 : 1;
                                                                                                                                                                                                                                                                                                                               ^

TypeError: infix.replace is not a function
    at TagExpressionParser.parse (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber-tag-expressions/lib/tag_expression_parser.js:9:24)
    at new ScenarioFilter (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/scenario_filter.js:14:52)
    at new HookDefinition (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/models/hook_definition.js:7:28)
    at Object.Before (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/support_code_library/helpers.js:40:910)
    at Object.module.exports (/Users/macmclaughlin/git/cucumber-api/lib/universe/features/internal_universe_test.js:56:8)
    at /Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/support_code_library/builder.js:60:2367
    at Array.forEach (native)
    at Object.build (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/support_code_library/builder.js:60:2334)
    at Cli.getSupportCodeLibrary (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/cli/index.js:43:32)
    at Cli._callee4$ (/Users/macmclaughlin/git/cucumber-api/node_modules/cucumber/lib/cli/index.js:48:43)
    at tryCatch (/Users/macmclaughlin/git/cucumber-api/node_modules/regenerator-runtime/runtime.js:63:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/macmclaughlin/git/cucumber-api/node_modules/regenerator-runtime/runtime.js:337:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/macmclaughlin/git/cucumber-api/node_modules/regenerator-runtime/runtime.js:96:21)
    at step (/Users/macmclaughlin/git/cucumber-api/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/macmclaughlin/git/cucumber-api/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
```

**Comment by charlierudolph** [comment]
I don't like allowing tags to remain as an array. For a lot of user input, I think it could be worthwhile to use something like [check-types](https://www.npmjs.com/package/check-types) to verify the type of arguments.

**Comment by charlierudolph** [comment]
Fix released in 2.0.0-rc.3
